formatter.go

```diff
@@ -21,6 +21,20 @@ type Formatter interface {
 	FormatFailure(*AssertionContext, *AssertionFailure) string
 }
 
+type FloatFormatMode int
+
+const (
+	// Print float in scientific notation for large exponent,
+	// otherwise print as decimal
+	FloatFormatAuto FloatFormatMode = iota
+
+	// Print float in decimal
+	FloatFormatDecimal
+
+	// Print float in scientific notation
+	FloatFormatScientific
```

Comment:
Let's refine comments a bit:

```
	// Print floats in scientific notation for large exponents,
	// otherwise print in decimal notation.
	// Similar to %g format.
	FloatFormatAuto FloatFormatMode = iota

	// Always print floats in decimal notation.
	// Similar to %f format.
	FloatFormatDecimal

	// Always print floats in scientific notation.
	// Similar to %e format.
	FloatFormatScientific
```